core/src/main/java/org/jdbi/v3/core/mapper/reflect/JdbiConstructors.java

```diff
@@ -44,6 +45,16 @@
             return annotatedConstructors.get(0);
         }
 
+        annotatedConstructors = Stream.of(constructors)
+                .filter(c -> c.isAnnotationPresent(ConstructorProperties.class))
+                .collect(Collectors.toList());
+
+        if (annotatedConstructors.size() > 1) {
+            throw new IllegalArgumentException(type + " must have at most one constructor annotated @ConstructorProperties");
+        } else if (annotatedConstructors.size() == 1) {
+            return annotatedConstructors.get(0);
+        }
+
         if (constructors.length != 1) {
             throw new IllegalArgumentException(type + " must have exactly one constructor, or specify it with @JdbiConstructor");
```

Comment:
This message should probably change too, e.g. `or specify it with @JdbiConstructor or @ConstructorProperties`.